src/OrleansRuntime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -153,7 +155,15 @@ private void InitializeGrainClassData(SiloAssemblyLoader loader, bool strict)
 
         private void InitializeInvokerMap(SiloAssemblyLoader loader, bool strict)
         {
-            IEnumerable<KeyValuePair<int, Type>> types = loader.GetGrainMethodInvokerTypes(strict);
+            // Load all existing implementations.
+            var existing = new HashSet<int>(loader.GetGrainMethodInvokerTypes(strict).Select(_ => _.Key));
+            
+            // Create implementations for all types which do not already have an implementation.
+            GrainMethodInvokerGenerator.CreateForCurrentlyLoadedAssemblies(type => !existing.Contains(type));
```

Comment:
From our last night's chat on Gitter.

I think (but I'm not sure) somewhere in AsseblyLoader may be a good place to invoke codegen. To logically make it part of discovering and loading of grain types.
Maybe at the end of LoadAssemblies()?

I think it would be best to check if there's already an invoker from compile time, log that fact, and not overwrite it with the generated one. Maybe even skip generation of invokers that are already loaded from compile time?
I think we want to be deterministic and explicitly log what invokers we found and generated. Should help debugging potential issues.